vortex-jni/src/array.rs

```diff
@@ -53,6 +56,105 @@ pub extern "system" fn Java_dev_vortex_jni_NativeArrayMethods_free(
     drop(unsafe { NativeArray::from_raw(array_ptr) });
 }
 
+#[unsafe(no_mangle)]
+pub extern "system" fn Java_dev_vortex_jni_NativeArrayMethods_exportToArrow<'local>(
+    mut env: JNIEnv<'local>,
+    _class: JClass,
+    array_ptr: jlong,
+    arrow_schema_ptr: JLongArray<'local>,
+    arrow_array_ptr: JLongArray<'local>,
+) {
+    // Return the arrow array here instead...I think?
+    let array_ref = unsafe { NativeArray::from_ptr(array_ptr) };
+
+    try_or_throw(&mut env, |env| {
+        let preferred_arrow_type = preferred_arrow_data_type(array_ref.inner.as_ref())?;
+        let viewless_arrow_type = data_type_no_views(preferred_arrow_type);
+
+        let arrow_array = array_ref.inner.clone().into_arrow(&viewless_arrow_type)?;
+        let Ok((ffi_array, ffi_schema)) = arrow::ffi::to_ffi(&arrow_array.to_data()) else {
+            throw_runtime!("FFI_ArrowSchema::try_from(&DataType) failed");
```

Comment:
I think the message is wrong, right? It's either exportToArrow failed or `to_ffi(&...)` failed